legion/engine/physics/physx/physx_event_process_funcs.cpp

```diff
@@ -127,4 +127,49 @@ namespace legion::physics
             instantiateNextCollider<PxSphereGeometry, float&>(getSDK(), wrapper, localTransform, sceneInfo, radius);
         }
     }
+
+    void processAddFirstConvex(physics_component& physicsComponent, const PhysxEnviromentInfo& sceneInfo, PhysxInternalWrapper& wrapper, ecs::entity entity)
+    {
+        PhysicsComponentData& data = physicsComponent.physicsCompData;
+
+        for (ConvexColliderData& convex : data.getConvexData())
+        {
+            if (convex.isRegisteredOfType(collider_type::quickhull_convex)) { continue; }
+
+            convex.setRegistered(true);
+
+            PxTransform transform;
+            PxTransform localTransform;
+
+            calculateGlobalAndLocalTransforms(localTransform, transform, convex, entity);
+            PxConvexMesh* convexMesh = static_cast<PxConvexMesh*>(convex.getConvexPtr().ptr);
+
+            if (wrapper.bodyType == physics_body_type::rigidbody)
```

Comment:
this should probably be a switch case, to set a precedence for handling later physics body types. also allows you to log warnings for unimplemented body types